File: go-sdk/pkg/models/operations/updatearticle.go
package operations

import (
	"github.com/speakeasy-sdks/forem-sdks/go-client-sdk/pkg/models/shared"
)

type UpdateArticlePathParams struct {
	ID int32 `pathParam:"style=simple,explode=false,name=id"`
}

type UpdateArticleRequest struct {
	PathParams UpdateArticlePathParams
	Request    *shared.Article `request:"mediaType=application/json"`
}

type UpdateArticleResponse struct {
	ContentType string
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/updatearticle.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class UpdateArticlePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: number;
}


export class UpdateArticleRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: UpdateArticlePathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.Article;
}


export class UpdateArticleResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
